### ISSUE TITLE
Run time error when model is not implemented via nn.Sequential

### ISSUE THREAD

**Issue by IdanAchituve** [AlexImmer/Laplace#92]
Hi,
Thanks for this great package!
I believe that there is a bug when using models that are not implemented via nn.Sequential. It also relates to the first issue reported  [here ](https://github.com/AlexImmer/Laplace/issues/90) which wasn't addressed.
Consider the following code snippet (basically taken from the toy example in this repo):

```
import torch
import torch.utils.data as data_utils
import torch.nn as nn
import torch.nn.functional as F

from laplace import Laplace


def get_sinusoid_example(n_data=150, sigma_noise=0.3, batch_size=150):
    # create simple sinusoid data set
    X_train = (torch.rand(n_data) * 8).unsqueeze(-1)
    y_train = torch.sin(X_train) + torch.randn_like(X_train) * sigma_noise
    train_loader = data_utils.DataLoader(
        data_utils.TensorDataset(X_train, y_train),
        batch_size=batch_size
    )
    X_test = torch.linspace(-5, 13, 500).unsqueeze(-1)
    return X_train, y_train, train_loader, X_test

# sample toy data example
X_train, y_train, train_loader, X_test = get_sinusoid_example(sigma_noise=0.3)

class FCNet(nn.Module):
    def __init__(self):
        super(FCNet, self).__init__()
        self.fc1 = nn.Linear(1, 50)
        self.fc2 = nn.Linear(50, 1)

    def forward(self, x):
        x = F.tanh(self.fc1(x))
        return self.fc2(x)

```

The following works correctly:
```
def get_model1():
    torch.manual_seed(711)
    return torch.nn.Sequential(
        torch.nn.Linear(1, 50), torch.nn.Tanh(), torch.nn.Linear(50, 1)
    )
model = get_model1()
la = Laplace(model, 'regression', subset_of_weights='all', hessian_structure='diag')
la.fit(train_loader)
```

Yet when changing to:
```
def get_model2():
    torch.manual_seed(711)
    return FCNet()

model = get_model2()
la = Laplace(model, 'regression', subset_of_weights='all', hessian_structure='diag')
la.fit(train_loader)
```

I get the following error: 
> NotImplementedError: Extension saving to diag_ggn_exact does not have an extension for Module <class '__main__.FCNet'>

The same happens with full hessian structure. 
Am I missing something?

Thanks,
Idan

**Comment by sungyubkim** [comment]
I am also having the same problem.

If I change lines 54~57 of `examples/calibration_example.py` as

```
la = Laplace(model, 'classification',
             subset_of_weights='all',
             hessian_structure='diag',
             )
````
I also got the following error code.

```
Traceback (most recent call last):
  File "calibration_example.py", line 58, in <module>
    la.fit(train_loader)
  File "/home/sungyub/Laplace/laplace/baselaplace.py", line 377, in fit
    loss_batch, H_batch = self._curv_closure(X, y, N)
  File "/home/sungyub/Laplace/laplace/baselaplace.py", line 960, in _curv_closure
    return self.backend.diag(X, y, N=N)
  File "/home/sungyub/Laplace/laplace/curvature/backpack.py", line 124, in diag
    loss.backward()
  File "/opt/conda/lib/python3.6/site-packages/torch/_tensor.py", line 307, in backward
    torch.autograd.backward(self, gradient, retain_graph, create_graph, inputs=inputs)
  File "/opt/conda/lib/python3.6/site-packages/torch/autograd/__init__.py", line 156, in backward
    allow_unreachable=True, accumulate_grad=True)  # allow_unreachable flag
  File "/opt/conda/lib/python3.6/site-packages/backpack/__init__.py", line 127, in hook_run_extensions
    backpack_extension.apply(module, g_inp, g_out)
  File "/opt/conda/lib/python3.6/site-packages/backpack/extensions/backprop_extension.py", line 86, in apply
    module_extension = self.__get_module_extension(module)
  File "/opt/conda/lib/python3.6/site-packages/backpack/extensions/backprop_extension.py", line 72, in __get_module_extension
    + "Module {}".format(module.__class__)
NotImplementedError: Extension saving to diag_ggn_exact does not have an extension for Module <class 'helper.wideresnet.WideResNet'>
```

Is there any simple method to solve this issue?

**Comment by runame** [comment]
Hi, the issue occurs when using `subset_of_weights='all'` and a model which is not implemented via `nn.Sequential` with a [BackPACK](https://github.com/f-dangel/backpack/) (which is the default) instead of an [ASDL](https://github.com/kazukiosawa/asdfghjkl) backend. So the solution in both of your cases should simply be:
```python
from laplace.curvature import AsdlGGN
...
la = Laplace(..., backend=AsdlGGN)
...
```
This was also addressed in #76. We will add a note in the docs clarifying the use of the backends.

**Comment by IdanAchituve** [comment]
Perfect, thanks.
